File: class2relational/src/main/java/org/atlanmod/model/dynamic/classModel/ClassElement.java
package org.atlanmod.model.dynamic.classModel;

import org.atlanmod.model.dynamic.DynamicElement;

public abstract class ClassElement extends DynamicElement {

    abstract String getId();

    public ClassElement(String classname) {
        super(classname);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof ClassElement){
            ClassElement obj_ = (ClassElement) obj;
            return this.getType().equals(obj_.getType()) && this.getId().equals(obj_.getId());
        }
        return false;
    }

}

File: sparktl-transformations/src/main/scala/org/atlanmod/class2relational/model/classmodel/ClassElement.scala
package org.atlanmod.class2relational.model.classmodel

import org.atlanmod.tl.model.impl.dynamic.DynamicElement

import scala.collection.mutable

abstract class ClassElement(classname: String) extends DynamicElement(classname, mutable.HashMap()) {

    def getId: String

    override def getType: String = classname

    override def equals(o: Any): Boolean = {
        o match {
            case obj: ClassElement =>
                this.getType.equals(obj.getType) && this.getId.equals(obj.getId)
            case _ => false
        }
    }

    override def weak_equals(o: Any): Boolean = {
        o match {
            case obj: ClassElement =>
                this.getType.equals(obj.getType)
            case _ => false
        }
    }

}
